### PR TITLE
Add more \l REPL commands.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add `\lm [PATTERN]` command for listing modules, optionally filtered by
name using the specified regexp pattern.
Add `\lmI [PATTERN]` command for case-sensitive pattern matching.

Add `\lr [PATTERN]` command for listing roles, also with option to
filter by name using regexp.
Add `\lrI [PATTERN]` command for case-sensitive pattern matching.

The PATTERN is purely regexp, it should not be quoted in any way, but
instead will be treated as a raw string for a regexp search.

Issue: #179.

Co-authored-by: Eduardo Schettino <schettino72@gmail.com>

### CODE REVIEW

**Comment by vpetrovykh** [comment]
Special thanks to @schettino72 on whose PR #672 this work is based.

File: edb/repl/utils.py [hunk 1]
```diff
@@ -124,3 +124,17 @@ def normalize_name(name: str) -> str:
...
+
+
+def get_filter_based_on_pattern(pattern: str,
+                                field: str = '.name',
+                                flag: str = '') -> str:
+    if flag:
+        flag = f'(?{flag})'
+
+    if pattern:
+        return f'FILTER re_test($${flag}{pattern}$$, {field})'
```
Comment (1st1) [inline edb/repl/utils.py]: `pattern` must be escaped. For example, your code will break if someone inputs `\l $$`. People can also do an EdgeQL injection this way. Make pattern a parameter of the query; pass flags as parameters.

Comment (vpetrovykh) [reply edb/repl/utils.py]: I've updated the code to use named query parameters and passing the string as a value to those.

File: edb/repl/__init__.py [hunk 2]
```diff
@@ -289,17 +290,85 @@ def command_connect(self, args: str) -> None:
...
+
+    @_command('lrI', R'\lrI [PATTERN]',
+              'list roles (case sensitive pattern match)')
+    def command_list_roles_i(self, args: str) -> None:
+        self._command_simplelist(
+            args.strip(),
+            query='SELECT name := sys::Role.name',
+            name_field='name',
+            item_name='roles',
+            case_sensitive=True,
```
Comment (1st1) [inline edb/repl/__init__.py]: To avoid code dup please follow what we did for the `command_describe_object` method

Comment (vpetrovykh) [reply edb/repl/__init__.py]: OK, I've used the same pattern for "I" flag versions of the specific `\l` commands.

**Pull request merged**